### ISSUE TITLE
Cannot create VMM: Cannot open /dev/kvm. Error: No such device (os error 19)

### ISSUE THREAD

**Issue by vladfaust** [firecracker-microvm/firecracker#1111]
I'm trying to run firecracker on a Tiny Core Linux inside VirtualBox. I've installed KVM on it, but apparently it does not create `/dev/kvm` device automatically. This it what I did:

```sh
> id
uid=1001(tc) gid=50(staff) groups=50(staff)
> sudo mknod /dev/kvm c 10 232
> sudo chown root:staff /dev/kvm
> sudo chmod g+w,o= /dev/kvm
> ls -l /dev/kvm
crw-rw---- 1 root staff 10, 232 /dev/kvm
```

Then I ensured that Firecracker can run on the machine with following commands:

```sh
> whoami
tc
> uname -m
x86_64
> uname -r
4.19.10-tinycore64
> [ -r /dev/kvm ] && [ -w /dev/kvm ] && echo "OK"
OK
> dmsg | grep -i "hypervisor"
Hypervisor detected: KVM
```

However, running firecracker without arguments gives the following error:

```sh
> whoami
tc
> ./firecracker-v0.16.0
... Firecracker panicked at 'Cannot create VMM: Cannot open /dev/kvm. Error: No such device (os error 19)', src/libcore/result.rs:1009:5
```

I totally understand that nested virtualization support is questionable. But I'm using VirtualBox on my Windows machine to develop locally to save some money on bare-metal servers.

Please, give me a hand here. It's very strange to receive such an error, because there certainly **is** `/dev/kvm` device (or, char) on my machine. Why firecracker does not see it? 🤔

**Comment by iggy** [comment]
Creating `/dev/kvm` yourself is certainly not the right way to start. When you load the kvm modules, it creates `/dev/kvm` itself. If it doesn't, it puts a message in `dmesg` to explain why. You should start by checking there.

**Comment by petreeftime** [comment]
Creating the node manually should be fine if the device actually exists. However, I don't think it does. Error 19 corresponds to `ENODEV`, which means the device doesn't exist at major=10 minor=232. Can you check if it shows up in `/proc/misc`? I should have the correct minor there if it exists. You can also check if `lsmod | grep kvm` shows the module loaded.

**Comment by vladfaust** [comment]
Thank you for your responses,

> Creating the node manually should be fine if the device actually exists.

That's what I needed to hear. So, the problem is that the module isn't loaded at all. Thanks to @petreeftime advice; `lsmod` has nothing mentioning `kvm`.

And yes, @iggy, `dmesg` didn't put anything useful regarding to KVM.

I've tested Ubuntu server with the same VirtualBox configuration and it works flawlessly. The problem is with TCL itself. There is the forum topic for reference: http://forum.tinycorelinux.net/index.php?action=profile;area=showposts;u=20429

I'm closing the issue for now as it's clearly out of Firecracker's scope. Thanks, @iggy, @petreeftime 🤗
